src/Client.php

```diff
@@ -42,6 +44,24 @@ public function users(): UsersClient
         );
     }
 
+    public function pages(): PagesClient
+    {
+        return new PagesClient(
+            $this->psrClient,
+            $this->token,
+            self::NOTION_VERSION
+        );
+    }
+
+    public function databases(): DatabasesClient
+    {
+        return new DatabasesClient(
+            $this->psrClient,
+            $this->token,
+            self::NOTION_VERSION,
+        );
+    }
+
     private static function resolvePsrClient(): ClientInterface
```

Comment:
I think this method is not really used, because you always use `nyholm/psr7`? 

It would perhaps be better to rely on something like http://httplug.io/, which already has its own (tested) client discovery logic, and you don't need to hardcode against unknown classes (which will be reported by `vimeo/psalm` as problematic)